### DISCUSSION TITLE
Can I sort struct methods using Rust LSP?

### DISCUSSION THREAD

**Discussion by wvffle** [saghen/blink.cmp#300] [Q&A] [answered]
First of all, the plugin is amazing! Thanks!

Now, to my question. Am I able to somehow sort the Rust's LSP completions?   

The screenshot below shows that the trait methods are shuffled within struct's methods which is something I don't want.

If I do not type anything after `pattern.`, I'd like to see struct fields first and struct methods second. Then after them, trait methods. 

Am I able to somehow do this?

![image](https://github.com/user-attachments/assets/db8267da-a591-4a51-9398-0b77d7982dba)

**Comment by saghen** [comment, marked as answer]
I've just closed #365 which uses the LSPs sorting hints. I've noticed for rust this now sorts the struct methods first so watch out for a release

> First of all, the plugin is amazing! Thanks!

You're very welcome :)

**Reply by saghen** [reply]
Fixed in v0.7.0
